### PR TITLE
Factor out a with_keys_renamed function.

### PR DESCRIPTION
**Pull request by toolness**
This is a simple readability improvement that makes our code a bit more DRY.  Right now the new function is only used in one place but I've had need of it in the past, so hopefully it will come in handy more than this once.

### CODE REVIEW

File: project/util/rename_dict_keys.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from typing import Any, Dict
+
+
+def with_keys_renamed(d: Dict[str, Any], renames: Dict[str, str]) -> Dict[str, Any]:
+    """
+    Returns a shallow copy of the given dictionary, with the given keys renamed, e.g.:
+
+        >>> with_keys_renamed({'foo': 1, 'bar': 2}, {'foo': 'baz'})
+        {'bar': 2, 'baz': 1}
```
Comment (toolness) [inline project/util/rename_dict_keys.py:9]: FYI this is a [doctest](https://docs.python.org/3/library/doctest.html)--its primarily usefulness is as documentation, but it's also executed as a test during CI to make sure it actually does what it says.

This is one of my favorite python builtins, though things can get a bit nutters once you start using this as the primary test suite for code, as it detracts from the readability of the documentation.  For this reason I usually limit doctests to just a handful of salient examples, while handling rare edge cases and such in a separate test suite.

Comment (samaratrilling) [reply project/util/rename_dict_keys.py:9]: This is neato

**Review by samaratrilling — APPROVED** [review]
